runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java

```diff
@@ -119,7 +124,11 @@ public ILogData peek(RuntimeLayout runtimeLayout, long globalAddress) {
 
         // Merge the read responses from different log unit servers
         Map<Long, LogData> readResult = futures.stream()
-                .map(future -> CFUtils.getUninterruptibly(future).getAddresses())
+                .map(future -> {
+                    ReadResponse readResponse = CFUtils.getUninterruptibly(future);
+                    updateCompactionMark(runtimeLayout.getRuntime(), readResponse.getCompactionMark());
```

Comment:
I think you missed another place to update compaction mark. readAll() in this class. readAll() is not only used for state transfer and fast loader, it's also used for stream layer, you need to update that. Also why don't you update compaction mark in LogUnitClient? I understand compactionMark is an object layer notation, but updating it in different places makes it error prone, one could easily forget to update when adding new APIs.

Reply:
I added compaction mark check to other stream view read.

Reply:
It maybe not a good place to update compaction mark. It is very difficult to pass the reference of the compaction mark to the logunit client. Moreover, the functionality of logunit client should keep clean and simple, whose whole responsibility is to communicate with LogUnit servers.